plugins/clients/apps-client/src/hopeit/apps_client/__init__.py

```diff
@@ -403,26 +403,23 @@ async def _parse_response(self, response, context: EventContext,
         Parses http response from external App, catching Unathorized errors
         and converting the result to the desired datatype
         """
-        if responses and response.status in responses:
+        try:
+            response_type = datatype if response.status == 200 else responses[response.status]  # type: ignore
             if response.content_type == 'text/plain':
                 data = {target_event_name: await response.text()}
             else:
                 data = await response.json()
-            response_type = responses[response.status]
             if isinstance(data, list):
                 return Payload.from_obj(data, list, item_datatype=response_type)  # type: ignore
             return [Payload.from_obj(data, response_type, key=target_event_name)]
-        if response.status == 200:
-            data = await response.json()
-            if isinstance(data, list):
-                return Payload.from_obj(data, list, item_datatype=datatype)  # type: ignore
-            return [Payload.from_obj(data, datatype, key=target_event_name)]
-        if response.status == 401:
-            raise Unauthorized(context.app_key)
-        if response.status >= 500:
-            raise ServerException(await response.text())
-        raise UnhandledResponse(f"Missing {response.status} status handler, use `responses` to handle this exception",
-                                await response.text(), response.status)
+        except (TypeError, KeyError):
```

Comment:
I don´t think it is safe to handle responses None case with a TypeError, it could hide any other Type error that happens in the lines of code between try and except. Probably will move the except block right after line 407 and put the remaining processing in a `finally`block. To prevent this to happen.

Also a add a line comment explaining the handling of TypeError case (i.e. when responses is None) and KeyError (when the status code is not found in responses)

Reply:
Fixed in [d71d9ec](https://github.com/hopeit-git/hopeit.engine/pull/136/commits/d71d9ecc8f3064fdc0402dae23fa22d1b4d928b6)